src/popup/pages/JwtSign.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <IonPage />
+</template>
+
+<script lang="ts">
+import { IonPage } from '@ionic/vue';
+import { defineComponent, onMounted } from 'vue';
+import { useRoute } from 'vue-router';
+import { Encoded, decode, signJwt } from '@aeternity/aepp-sdk';
+
+import { AeAccountHdWallet } from '@/protocols/aeternity/libs/AeAccountHdWallet';
+import { JWT_HEADER, MODAL_CONFIRM_UNSAFE_SIGN, PROTOCOLS } from '@/constants';
+import { handleUnknownError, toBase64Url } from '@/utils';
+import { RejectedByUserError } from '@/lib/errors';
+import {
+  useAccounts,
+  useAeSdk,
+  useDeepLinkApi,
+  useModals,
+  useUi,
+} from '@/composables';
+
+export default defineComponent({
+  name: 'JwtSign',
+  components: {
+    IonPage,
+  },
+  setup() {
+    const route = useRoute();
+
+    onMounted(async () => {
+      const { callbackOrigin, openCallbackOrGoHome } = useDeepLinkApi();
+      const { nodeNetworkId } = useAeSdk();
+      const { openModal } = useModals();
+      const { setLoaderVisible } = useUi();
+      const { getLastActiveProtocolAccount } = useAccounts();
+
+      try {
+        setLoaderVisible(true);
+        const payload = route.query.payload?.toString();
+        const jwk = route.query.jwk?.toString() === 'true';
+        const { host, href, protocol } = callbackOrigin.value || {} as any;
+
+        if (!payload) {
+          throw new Error('There is no payload to sign');
+        }
+
+        const payloadAsJson = JSON.parse(payload);
+        if (jwk) {
+          payloadAsJson.sub_jwk = {
+            kty: 'OKP',
+            crv: 'Ed25519',
+            x: toBase64Url(decode(
+              getLastActiveProtocolAccount(PROTOCOLS.aeternity)?.address as Encoded.AccountAddress,
+            )),
+          };
+        }
+
+        const messageToConfirm = `${JWT_HEADER}.${toBase64Url(JSON.stringify(payloadAsJson))}`;
+
+        await openModal(MODAL_CONFIRM_UNSAFE_SIGN, {
+          data: messageToConfirm,
+          app: {
+            host,
+            protocol,
+            name: host,
+            url: href,
+          },
+        });
+
+        const dataToSign = JSON.parse(payload);
+
+        if (!jwk) {
+          dataToSign.sub_jwk = undefined;
+        }
+
+        const signature = await signJwt(dataToSign, new AeAccountHdWallet(nodeNetworkId));
+        openCallbackOrGoHome(true, { signature: signature as any });
```

Comment:
previously it was mentioned by @davidyuk that "signature" is semantically incorrect, as it is includes the payload, so I opted to use "signed-payload" https://github.com/superhero-com/superhero-wallet/pull/2875/files#diff-663c4a09a9d8aef6a1de5c7e9e3b4d84d60e4f9bba55c86b530b2837d8c09d39R58